apps/meteor/ee/server/apps/communication/rest.ts

```diff
@@ -1120,24 +1120,57 @@ export class AppsRestApi {
 					return API.v1.notFound(`No App found by the id of: ${this.urlParams.id}`);
 				},
 				async post() {
-					if (!this.bodyParams.status || typeof this.bodyParams.status !== 'string') {
+					const { id: appId } = this.urlParams;
+					const { status } = this.bodyParams;
+
+					if (!status || typeof status !== 'string') {
 						return API.v1.failure('Invalid status provided, it must be "status" field and a string.');
 					}
 
-					const prl = manager.getOneById(this.urlParams.id);
-
+					const prl = manager.getOneById(appId);
 					if (!prl) {
-						return API.v1.notFound(`No App found by the id of: ${this.urlParams.id}`);
+						return API.v1.notFound(`No App found by the id of: ${appId}`);
 					}
 
-					if (AppStatusUtils.isEnabled(this.bodyParams.status)) {
-						if (!(await canEnableApp(prl.getStorageItem()))) {
-							return API.v1.failure('Enabled apps have been maxed out');
+					const storedApp = prl.getStorageItem();
+					const marketplaceInfo = storedApp.marketplaceInfo;
+					let isAppEnterpriseOnly = !!marketplaceInfo?.isEnterpriseOnly;
+
+					if (!isEnterprise()) {
+						const { version } = prl.getInfo();
+						const baseUrl = orchestrator.getMarketplaceUrl();
+						const headers = getDefaultHeaders();
+
+						const appInfosURL = new URL(`${baseUrl}/v1/apps/${appId}`);
+						appInfosURL.searchParams.set('appVersion', String(version));
+
+						try {
+							const appInfoResponse = await fetch(appInfosURL.toString(), {
```

Comment:
This request fails if the app is not found in the marketplace-api. This means calls to this endpoint for private apps will not work, as pointed out before https://github.com/RocketChat/Rocket.Chat/pull/29683#discussion_r1256327089